### ISSUE TITLE
Doesn't Store Special Characters

### ISSUE THREAD

**Issue by AlphaLeoli** [70CentsApple/ChatNotifier#31]
### Bug description

The responder works perfectly fine when you put in a unicode character. However, after you relaunch it changes to a question mark. This happens both in the regex (which makes it so you can't make it respond to people when they send a special character) and in the actual response (which makes you look confused all the time when you send the question mark)

### Steps to reproduce

1. make a responder to respond with a special unicode character like the one i used, `💕`
2. use it, to make sure it works
3. relaunch the game
4. use it and it will appear as question marks.

### Expected behavior

![image](https://github.com/user-attachments/assets/e8af580b-43a1-4247-a1be-b8f27993867f)

### Actual behavior

![image](https://github.com/user-attachments/assets/69047fa1-8c23-4596-93e6-582b351fc2ce)

### Minecraft version

1.20.4

### Mod version

2.2.1

### Check list

- [X] I have verified that the issue persists in the latest version of the mod.
- [X] I have searched the existing issues and confirmed that this is not a duplicate.

**Comment by 70CentsApple** [comment]
I couldn't reproduce it in my dev environment. How about trying using [the latest artifact](https://github.com/70CentsApple/ChatTools/actions/runs/9919297403) and tell if the problem still exists?

**Comment by AlphaLeoli** [comment]
Yeah it still happens. I will show the steps to reproduce in more detail
1. go into the game and create a filter. the one i have is this
![image](https://github.com/user-attachments/assets/5cb2d058-3fdd-4cfd-a0e3-4c77a68477a5)
2. make sure it works. for me, it did! 
![image](https://github.com/user-attachments/assets/30fcce05-d34f-4f60-a4cc-b87754f2d016)
3. after testing, alt+f4 your game, and then launch back up
4. go into the config, and you will see that the hearts turned into question marks. As you can see in this screenshot, all my other triggers that have characters like that are question marks and no longer work properly.
![image](https://github.com/user-attachments/assets/033ebee1-28a1-4b00-95ab-2e1c18bbeb27)

**Comment by 70CentsApple** [comment]
Still, can't reproduce.
Could you please do `/chattools config openfile`, and tell me the encoding?
It can be seen here ↓
![image](https://github.com/user-attachments/assets/670cc5e0-110b-4fb4-abc9-f508b1755663)

**Comment by AlphaLeoli** [comment]
UTF-8

**Comment by 70CentsApple** [comment]
> UTF-8

It is supposed to work fine, but on your computer it doesn't. That's weird.

**Comment by 70CentsApple** [comment]
@AlphaLeoli Hey! Does this problem still exist?

**Comment by AlphaLeoli** [comment]
i will check in a bit am doing smt

**Comment by AlphaLeoli** [comment]
oh i almost forgot to do this
but yes it still happens

**Comment by tmmovo** [comment]
I think the best way to solve this is to directly write its corresponding Unicode encoding in the source file (it can't be done in-game, it will be escaped), so it shouldn't appear as a problem when re-entering......it might be a bit troublesome?
